precompiles/staking/logs.go

```diff
@@ -0,0 +1,154 @@
+package staking
+
+import (
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/vm"
+)
+
+func (c *Contract) AddStakeLog(
+	ctx sdk.Context,
+	stateDB vm.StateDB,
+	staker common.Address,
+	validator string,
+	amount *big.Int,
+) error {
+	event := c.Abi().Events["Stake"]
```

Comment:
Can we use delegate ? I think cosmos referees to stake as the creation of a validator

Reply:
We decieded to go with stake/unstake/moveStake terminology for the staking precompile